unit-tests/log/test-multi-thread-logs.cpp

```
@@ -0,0 +1,221 @@
+            std::chrono::milliseconds delta_ms = ms - start_ms;
+            if (delta_ms > max_time) max_time = delta_ms;
+            if (delta_ms < min_time) min_time = delta_ms;
+            avg_time += delta_ms;
+            start_ms = ms;
+            ++iterations;
+        }
+    };
+    const char* log_file_path = ".//multi-thread-logs.log";
+    try {
+        remove(log_file_path);
+    }
+    catch (...) {}
+    rs2::log_to_file(RS2_LOG_SEVERITY_DEBUG, log_file_path);
+
+    std::vector<std::thread> threads;
+
+    for (int i = 0; i < 10; ++i)
+    {
+        threads.push_back(std::thread(func, i + 1));
+    }
+    
+    for (auto&&t : threads)
+    {
+        if (t.joinable()) t.join();
+    }
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(5000));
+    std::stringstream ss;
+    ss << "max time = " << max_time.count() << " ms" << std::endl;
+    ss << "min time = " << min_time.count() << " ms" << std::endl;
+    ss << "avg time = " << (float) (avg_time.count()) / (number_of_iterations * 10.f) << " ms" << std::endl;
+    CAPTURE(ss.str());
+    CHECK_NOFAIL(max_time.count() < 10);
+    REQUIRE(max_time.count() < 50);
+    
+    // checking logs correctness - that all the integer's values have been logged
+    {
+        // preparing array of number of iterations * number of threads
+        std::array<char, number_of_iterations * 10> check_table;
+        // setting all cells to 0
+        memset(&check_table[0], 0, check_table.size());
+        // marking cells at the position of each log integer value to 1
+        std::ifstream check_file(log_file_path, std::ios::in);
+        if (check_file.good())
+        {
+            std::string line;
+            while (std::getline(check_file, line)) {
+                auto value_in_log = stoi(line.substr(line.find_last_of(" ")));
+                check_table.at(value_in_log - 1) = 1;
```

Add try/catch to avoid possible segfault if the value read from the file is bigger than the array size

---

is empty catch enough? otherwise, what do you propose to do in it? fail the test?